atlasdb-api/src/main/java/com/palantir/atlasdb/metrics/FilteredTaggedMetricSet.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.metrics;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+import com.codahale.metrics.Metric;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricSet;
+
+public class FilteredTaggedMetricSet implements TaggedMetricSet {
+    private final TaggedMetricSet unfiltered;
+    private final Map<MetricName, List<MetricPublicationFilter>> singleMetricFilters;
```

Comment:
Would it make sense for this class to just take a Predicate<MetricName>, and implement the filtering rules somewhere else?

Reply:
As in this class just takes `Predicate<MetricName>` (and then have some kind of filter registry that handles this map)? Yep, we could - I don't feel strongly either way (what you propose does split responsibilities better, though it does add another piece).

Reply:
(for clarity: yeah I plan to change this)